### ISSUE TITLE
Better experience when many projects are available to import

### ISSUE THREAD

**Issue by takluyver** [rtfd/readthedocs.org#1899]
I like the 'import project' design introduced in #1695 overall, but the pagination is annoying me. I'm in several organisations with many repositories, and as my username is relatively late in the alphabet, I have to click through many pages to get to my own personal repos.

How could this be better:
- I can filter down to an organisation's repos easily, but not to my own personal repos (takluyver/foo). I guess this would be simple to add.
- Put a search box on the import project page, preferably with search-as-you-type.
- Sort recently updated repos to the top, like Github's [repo view](https://github.com/takluyver?tab=repositories), as they're the ones I'm most likely to want to use. I'm baffled by the sort order at the moment - it seems mostly alphabetical, with some things scattered randomly into it.

**Comment by agjohnson** [comment]
Yes, sort is alphabetical, with grouping by organization. This pagination is a first pass, the second pass is to implement a fuzzy search, keeping pagination. We can't sort on most used from github, we don't have the information to make that decision.

**Comment by agjohnson** [comment]
For anyone willing to take this work on, it should be a mostly digestible chunk. The front end interaction is entirely tied to the API results and pagination. So we'll want to offer fuzzy search through the API. Pagination will stay the same, but will also operate on a filtered list of elements. This would require few changes on the front end side, minus the addition of a search box that triggers a fuzzy search on after a delay from typing or something.

**Comment by adrn** [comment]
+1 to this

A minimal addition to get all repos that don't belong to an organization would be incredibly helpful. I'm currently in a state where I can't use the automatic import because I'm in the conda-forge organization and I have 10s of pages of repos that make finding my own projects tedious and impractical.

**Comment by humitos** [comment]
Related to #3337

**Comment by ericholscher** [comment]
Seems with a lot of repos, this is now timing out, so something we should re-prioritize.

**Comment by agjohnson** [comment]
What specifically is timing out?

That might be something to address separately if it is a problem with data modeling, as this is discussing the UI interaction. If it's a matter of our indexes being poorly defined for how we are querying, then adding fuzzy search could turn this slow query into a 100% unusable query. I'd rather go the route of fuzzy search for our v2 of this feature. Perhaps we disregard the need for pagination for our next implementation -- especially if it adds to query speed slow down, as pagination is always going to be slow.

On the data modeling side we also have an issue open somewhere about redefining this modeling, so that `RemoteRepository` instances are defined once, and have a many-to-many relation to the remote user instance. So, instead of there being a RemoteRepository for `rtfd/readthedocs.org` repository for everyone in our org, there is one `RemoteRepository` instance that is shared. This reduces table bloat, but also makes managing webhooks make more sense.

**Comment by WoLpH** [comment]
The timing out is now fixed luckily :)
https://github.com/rtfd/readthedocs.org/issues/5441

But it's definitely not ideal for people like me with too many repos. Even when just looking through my own repos I've got 14 pages to browse through. I think that even a simple `repo_name__istartswith` would be more than sufficient. Assuming that in the case of this repo it would search in the `readthedocs.org` portion and not the entire `rtfd/readthedocs.org`, not sure if that field is available.
A full `icontains` would most likely be very slow if you have the 6400 conda-forge packages in the list of available options.
